src/bcs/generate_bc_eqs.jl

```diff
@@ -1,119 +1,189 @@
+# max of the count for each depvar in the system of equations.
+@inline function get_bc_counts(i, s, bcs)
+    left = 0
+    right = 0
+    for u in s.ū
+        depvaredges = get_depvarbcs(u, s, i)
+        counts = [map(u -> occursin(u, bc.lhs), depvaredges) for bc in bcs]
+        left = max(left, sum([c[1] for c in counts]))
+        right = max(right, sum([c[2] for c in counts]))
     end
+    return [left, right]
+end
 
+
+"""
+Mutates bceqs and u0 by finding relevant equations and discretizing them.
+TODO: return a handler for use with generate_finite_difference_rules and pull out initial condition. Important to remember that BCs can have 
+"""
+function BoundaryHandler!!(u0, bceqs, bcs, s::DiscreteSpace, depvar_ops, tspan, derivweights::DifferentialDiscretizer) 
+    
+    t=s.time
+    
     if t === nothing
-        initmaps = s.vars
+        initmaps = s.ū
     else
-        initmaps = substitute.(s.vars,[t=>tspan[1]])
+        initmaps = substitute.(s.ū,[t=>tspan[1]])
     end
 
-    # All unique order of derivates in BCs
-    bc_der_orders = filter(!iszero,sort(unique([count_differentials(bc.lhs, s.nottime[1]) for bc in bcs])))
-    # no. of different orders in BCs
-    n = length(bc_der_orders)
-    return BoundaryMap(depvarbcmaps, DerivativeMap{n}(depvarderivbcmaps, bc_der_orders), initmaps)
-end
- 
-### INITIAL AND BOUNDARY CONDITIONS ###
-"""
-Mutates bceqs and u0 by finding relevant equations and discretizing them
-"""
-function generate_u0_and_bceqs!!(u0, bceqs, bcs, s, depvar_ops, tspan, discretization)
-    grid_align = discretization.grid_align
-    t=discretization.time
-    boundarymap = generate_maps(bcs, t, s, tspan, grid_align)
-      # Generate initial conditions and bc equations
+    # Create some rules to match which bundary/variable a bc concerns
+    # * Assume that the term of the condition is applied additively and has no multiplier/divisor/power etc.
+    
+    ## BC matching rules, returns the variable and parameter the bc concerns
+
+    lower_boundary_rules, upper_boundary_rules = generate_boundary_matching_rules(s, derivweights.orders)
+
+    # indexes for Iedge depending on boundary type
+    idx(::LowerBoundary) = 1
+    idx(::UpperBoundary) = 2
+    Iedge = edges(s)
+    # Generate initial conditions and bc equations
     for bc in bcs
+        # * Assume in the form `u(...) ~ ...` for now
         bcdepvar = first(get_depvars(bc.lhs, depvar_ops))
-        if any(u->isequal(operation(u),operation(bcdepvar)), s.vars)
+        
+        if any(u -> isequal(operation(u), operation(bcdepvar)), s.ū)
             if t !== nothing && operation(bc.lhs) isa Sym && !any(x -> isequal(x, t.val), arguments(bc.lhs))
                 # initial condition
-                # Assume in the form `u(...) ~ ...` for now
-                initindex = findfirst(isequal(bc.lhs), boundarymap.initial) 
+                # * Assume that the initial condition is not in terms of the initial derivative
```

Comment:
this comment is not related to the one before? The code looks like it has the same limitation?